src/lib/routers/history.ts

```diff
@@ -78,6 +79,13 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
    */
   private shouldPushState: boolean = true;
 
+  /**
+   * Indicates if `write` should be called on `dispose`.
+   *
+   * @default true
+   */
+  private writeOnDispose: boolean = true;
```

Comment:
I'd like to find a better name, but can't think of anything

- cleanup: true/false
- cleanupOnDispose: true/false
- singlepage: false/true (not really a fan)
- externalRouter: false/true (not really a fan)

Reply:
@Haroenv What don't you like about it? What meaning should it convey that you think is missing right now?

Reply:
for me it seems an implementation detail, I'd rather have the option explain why you'd want to do that in the first place

Reply:
Should we add a paragraph in the docs to explain this new option (and keep the name as is it)?